Select default items

joelfmrodrigues opened the pull request:
| Q               | A
| --------------- | ---

| New feature?    | [ X]

| Related issues?  |  mentioned in #8 

#### What's in this Pull Request?

Please describe the changes in this PR. Sample description or details around bugs which are being fixed.


Added new property to allow an array of default items to be passed to the component.
Added a new function that will select the items passed by default.
Calling the function on componentDidUpdate, so the items are selected and persisted if the render method is executed

This allow for simple scenarios like just selecting items by default when the page renders, but also more complex scenarios, like cascading lists (select items on the first list that will trigger selection of items on the second list)

estruyf commented:
Thanks for this PR. Just one remark. Wouldn't it be better to pass an array of item IDs instead of the actual item? That would make it easier to set the default selected items by user input.

joelfmrodrigues commented:
@estruyf I will try to change this over the next days and update the PR

joelfmrodrigues commented:
@estruyf  I have been unable to pick this up due to work...
Are you making the assumption that the items will always have an Id property? Or am I missing something?

joelfmrodrigues commented:
Hi @estruyf , I have now updated this to select items by index as it looked like the most sensible option to me so that it mirrors Details List functionality. Let me know if you still prefer an Id based implementation.
Additionally, I have added a tslint.json file at the root to allow the TSLint extension for VSCode to work with the rules on the config folder

estruyf commented:
Thanks, this is now merged to the dev branch.
